components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/SubProperty.java

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.common.model;
+
+import org.apache.axiom.om.OMElement;
+import org.apache.commons.collections.CollectionUtils;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlElement;
+import javax.xml.bind.annotation.XmlElementWrapper;
+import javax.xml.bind.annotation.XmlRootElement;
+
+/**
+ * Model class to store sub-properties of {@link Property} model.
+ */
+@XmlAccessorType(XmlAccessType.FIELD)
+@XmlRootElement(name = "SubProperty")
+public class SubProperty implements Serializable {
```

Comment:
Can you explain the requirement for a SubProperty with an example maybe?

Reply:
In the case of SAML Federated Authenticator, a single property can have sub properties. E.g is shown below.

Enable Artifact Binding
     Artifact Resolve Endpoint Url:
     Enable Artifact Resolve Request Signing
     Enable Artifact Response Signing